x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Verifier.java

```diff
@@ -212,6 +238,33 @@ private static Failure fail(Node<?> source, String message, Object... args) {
                 failures.addAll(localFailures);
             });
         }
+        
+        // gather metrics
+        if (failures.isEmpty() && metrics != null) {
```

Comment:
Can metrics ever be null? Isn't this class always available inside PlanExecutor?